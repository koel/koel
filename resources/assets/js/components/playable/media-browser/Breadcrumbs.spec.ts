import { describe, expect, it } from 'vitest'
import { createHarness } from '@/__tests__/TestHarness'
import Component from './Breadcrumbs.vue'

describe('breadcrumbs.vue', () => {
  const h = createHarness()

  it.each([[''], ['/var'], ['/var/media/'], ['/var/media/deep/nested/path']])('renders', path => {
    const { html } = h.render(Component, {
      props: {
        path,
      },
    })

    expect(html()).toMatchSnapshot()
  })
})
