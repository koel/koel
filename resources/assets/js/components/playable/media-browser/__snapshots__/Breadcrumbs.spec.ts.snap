// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`breadcrumbs.vue > renders 1`] = `
<ul data-v-a48b9e35="" class="text-base">
  <li data-v-a48b9e35="" class="inline-block"><a data-v-a48b9e35="" href="/#/browse" class="text-k-text-secondary font-normal">Library</a></li>
</ul>
`;

exports[`breadcrumbs.vue > renders 2`] = `
<ul data-v-a48b9e35="" class="text-base">
  <li data-v-a48b9e35="" class="inline-block"><a data-v-a48b9e35="" href="/#/browse" class="text-k-text-secondary font-normal">Library</a></li>
  <li data-v-a48b9e35="" class="inline-block"><a data-v-a48b9e35="" href="/#/browse/var" class="text-k-text-secondary font-normal">var</a></li>
</ul>
`;

exports[`breadcrumbs.vue > renders 3`] = `
<ul data-v-a48b9e35="" class="text-base">
  <li data-v-a48b9e35="" class="inline-block"><a data-v-a48b9e35="" href="/#/browse" class="text-k-text-secondary font-normal">Library</a></li>
  <li data-v-a48b9e35="" class="inline-block"><a data-v-a48b9e35="" href="/#/browse/var" class="text-k-text-secondary font-normal">var</a></li>
  <li data-v-a48b9e35="" class="inline-block"><a data-v-a48b9e35="" href="/#/browse/var/media" class="text-k-text-secondary font-normal">media</a></li>
</ul>
`;

exports[`breadcrumbs.vue > renders 4`] = `
<ul data-v-a48b9e35="" class="text-base">
  <li data-v-a48b9e35="" class="inline-block"><a data-v-a48b9e35="" href="/#/browse" class="text-k-text-secondary font-normal">Library</a></li>
  <li data-v-a48b9e35="" class="inline-block"><span data-v-a48b9e35="">â€¦</span></li>
  <li data-v-a48b9e35="" class="inline-block"><a data-v-a48b9e35="" href="/#/browse/var/media/deep/nested" class="text-k-text-secondary font-normal">nested</a></li>
  <li data-v-a48b9e35="" class="inline-block"><a data-v-a48b9e35="" href="/#/browse/var/media/deep/nested/path" class="text-k-text-secondary font-normal">path</a></li>
</ul>
`;
